pkg/clustermesh/mcsapi/metrics.go

```diff
@@ -0,0 +1,93 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package mcsapi
+
+import (
+	"context"
+	"log/slog"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	mcsapiv1alpha1 "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+
+	operatorMetrics "github.com/cilium/cilium/operator/metrics"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+	"github.com/cilium/cilium/pkg/metrics"
+)
+
+func registerMCSAPICollector(logger *slog.Logger, client client.Client) {
+	operatorMetrics.Registry.MustRegister(&mcsAPICollector{
+		logger: logger,
+		client: client,
+
+		serviceExportInfo: prometheus.NewDesc(
+			prometheus.BuildFQName(metrics.CiliumOperatorNamespace, "", "serviceexport_info"),
+			"Information about ServiceExport in the local cluster",
+			[]string{"serviceexport", "namespace"}, nil),
```

Comment:
Should the label be simply `name`, instead of `serviceexport`, if we end up keeping it? Similarly below.

Reply:
Well kube-state-metrics seems to prefer specific kind as name, I think it's when you need to "combine" metrics it's more type safe than if everything has the "name" label. But from the current metric there's probably not much need to do those kind of things maybe in the future though 🤷‍♂️